scipy/stats/_morestats.py

```diff
@@ -1311,28 +1339,33 @@ def _all(x):
         message = ("The `optimizer` argument of `boxcox_normmax` must return "
                    "an object containing the optimal `lmbda` in attribute `x`.")
         raise ValueError(message)
-    else:
-        # Test if the optimal lambda causes overflow
-        x = np.asarray(x)
-        x_treme = np.max(x, axis=0) if np.any(res > 0) else np.min(x, axis=0)
-        istransinf = np.isinf(special.boxcox(x_treme, res))
-        dtype = x.dtype if np.issubdtype(x.dtype, np.floating) else np.float64
-        if np.any(istransinf):
-            warnings.warn(
+    elif not np.isinf(ymax):  # adjust the final lambda
+        # x > 1, boxcox(x) > 0; x < 1, boxcox(x) < 0
+        xmax, xmin = np.max(x), np.min(x)
+        if xmin >= 1:
+            x_treme = xmax
+        elif xmax <= 1:
+            x_treme = xmin
+        else:  # xmin < 1 < xmax
+            indicator = special.boxcox(xmax, res) > abs(special.boxcox(xmin, res))
+            x_treme = xmax if np.any(indicator) else xmin
+
+        sign_x_treme_m1 = np.sign(x_treme - 1)
+        mask = special.boxcox(x_treme, res) * sign_x_treme_m1 > ymax
```

Comment:
IIUC, this should be equivalent to 
```suggestion
        mask = abs(special.boxcox(x_treme, res)) > ymax
```
If that's right, I think that would be a bit easier to reason about and a hair faster. But mostly I want to check my understanding.

Reply:
Thanks, this is correct.